src/Networking/NexusMods.Networking.Downloaders/DownloadService.cs

```diff
@@ -209,8 +223,17 @@ public async Task StopAsync(CancellationToken cancellationToken)
             // TODO(Al12rs): should Suspend() instead, but only after moving ongoing dl files outside Temp folder,
             // that is otherwise cleaned up on application close, causing exceptions due to file in use,
             // on top of discarding all the progress.
-            .Select(dl => dl.Cancel());
+            .Select(dl => dl.Suspend());
         
         await Task.WhenAll(suspendingTasks);
     }
+    
+    /// <summary>
+    /// Set a custom downloadDirectory, for tests only.
+    /// Directory should already exist.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal void SetDownloadDirectory(AbsolutePath path)
+    {
+        _downloadDirectory = path;
+    }
```

Comment:
Why is this needed?

Reply:
For tests, to allow each test to have a separate download location. 
This solution sucks. It would be much better if each test had its own `DownloadService` and entire service provider since we are already there.

Reply:
`OverrideForTests` doesn't fit this use-case?

Reply:
It would be the same for methods in the same test class, since singletons injected in the constructors are shared, from what I understand.

Reply:
Do these tests run in parallel? If not, is there an issue with using the default location?

Reply:
I don't know tbh, they way we set it up with a separate guid named folder for each task makes it so that each task is pretty much isolated, but if we ever add logic to clean up orphaned folders or something like that then this could break down.

Reply:
Ideally these tests would run all in parallel and they would each have a separate `DownloadService`, each with a different `DownloadSettings` override.

Reply:
Not with how we currently run tests.